docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,332 @@
+
+## File Structure
+
+Follow these guidelines to keep your test structure manageable and "clean":
+
+* Use a page model to store Selectors and compound actions that are used often across your app. For instance, put all actions that are necessary to perform an action into one reusable function.
+
+* Put all the page model files into one directory. If your application is divided logically into components or subsystems, split up the associated page model objects into separate files.
+
+> You can find an example of a page model in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+* Create a `testcaferc.json` file in the root directory of the project. It enables you to fine-tune your tests, which can be useful in CI/CD systems. For more information, read the [Configuration File](../../reference/configuration-file.md) article.
+
+* Define one `fixture` in every test file. While it's technically possible to define multiple, it may lead to confusion as your suite grows.
+
+* A `fixture`s should aggregate tests that relate to each other by meaning or testing scenario.
+
+* TestCafe tests are purely functional. As such, they should not depend on the implementation details and it is best to isolate them from production code. Keep your test files in a separate directory. You can name this directory appropriately (for instance, `tests`).
+
+* In your test folder, create subfolders for tests that cover different subsystems of your application.
+
+* Don't write long tests. Shorter test scenarios are easier to debug and can run concurrently.
+
+* Any reused data (for example, large sets of reference values or forms inputs) is better stored in a dedicated directory. Consider a descriptive folder name (for instance, `data`).
+
+With all the suggestions applied, your project's file structure might look like this:
+
+```sh
+.
+├── .testcaferc.json
+└── tests
+    ├── |- test_group1/
+    │   └── |-test1.js
+    │       |-test2.js
+    ├── |- test_group2/
+    │   └── |-test1.js
+    │       |-test2.js
+    ├── |- page_model/
+    │   └── |- page1.js
+    │       |- page2.js
+    ├── |- helpers/
+    │   └── |- helper1.js
+    │       |- helper2.js
+    └── |-data
+```
+
+## Setup and Teardown
+
+State management is an integral and important part of web testing. When your tests run, there are inevitably leftovers- database or local storage records, cache, or cookies. This data requires deletion during teardown. Extra steps your app requires to run (like adding records to the database) are performed during setup.
+
+One way to deal with this is to place the setup actions in the preceding test's `after` and `afterEach` hooks. While good for cleanup, these hooks create mutual dependence between your tests when used to set up for the following test. The success of a test is then influenced by a preceding one, which is not desirable.
```

Comment:
> when used to set up for the following test.

*for a following test* or *for a subsequent test*. Because I would expect a colon and some code after `for the following test`.